packages/cardano-services/test/StakePoolSeach/StakePoolSearchHttpServer.test.ts

```diff
@@ -0,0 +1,422 @@
+        new Pool({ connectionString: process.env.DB_CONNECTION_STRING })
+      );
+      stakePoolSearchHttpServer = StakePoolSearchHttpServer.create({ stakePoolSearchProvider }, config);
+      await stakePoolSearchHttpServer.initialize();
+      await stakePoolSearchHttpServer.start();
+    });
+
+    afterAll(async () => {
+      await stakePoolSearchHttpServer.shutdown();
+    });
+
+    describe('/health', () => {
+      it('forwards the stakePoolSearchProvider health response', async () => {
+        const res = await got(`${apiUrlBase}/health`, {
+          headers: { 'Content-Type': APPLICATION_JSON }
+        });
+        expect(res.statusCode).toBe(200);
+        expect(JSON.parse(res.body)).toEqual({ ok: true });
+      });
+    });
+
+    describe('/search', () => {
+      it('returns a 200 coded response with a well formed HTTP request', async () => {
+        expect(
+          (
+            await got.post(`${apiUrlBase}/search`, {
+              json: { args: [] }
+            })
+          ).statusCode
+        ).toEqual(200);
+      });
+
+      it('returns a 400 coded response if the wrong content type header is used', async () => {
+        try {
+          await got.post(`${apiUrlBase}/search`, {
+            headers: { 'Content-Type': APPLICATION_CBOR }
+          });
+          throw new Error('fail');
+        } catch (error: any) {
+          expect(error.response.statusCode).toBe(400);
+          expect(error.message).toBe(BAD_REQUEST_STRING);
+        }
+      });
+
+      describe('pagination', () => {
+        it('should paginate response', async () => {
+          const req = {};
+          const reqWithPagination = { pagination: { limit: 2, startAt: 1 } };
+          const response = await doServerRequest(req);
+          const responseWithPagination = await doServerRequest(reqWithPagination);
+          expect(response.stakePools.length).toEqual(8);
+          expect(responseWithPagination.stakePools.length).toEqual(2);
+          expect(response.stakePools[0]).not.toEqual(responseWithPagination.stakePools[0]);
+        });
+        it('should paginate response with or condition', async () => {
+          const req = { filters: { _condition: 'or' } };
+          const reqWithPagination = { ...req, pagination: { limit: 2, startAt: 1 } };
+          const response = await doServerRequest(req);
+          const responseWithPagination = await doServerRequest(reqWithPagination);
+          expect(response.stakePools.length).toEqual(8);
+          expect(responseWithPagination.stakePools.length).toEqual(2);
+          expect(response.stakePools[0]).not.toEqual(responseWithPagination.stakePools[0]);
+        });
+      });
+
+      describe('search pools by identifier filter', () => {
+        // response should be the same despite the high order condition
+        it('or condition', async () => {
+          const req = {
+            filters: {
+              _condition: 'or',
```

Comment:
```suggestion
```
It's confusing when there's just a single filter. If we want to demonstrate the query is still valid, make an explicit one.